### PR TITLE
fix(coverage): get close to 100% test coverage

### PR DESCRIPTION
**Pull request by SimenB**
It would be great if we could get to 100% coverage, especially as we're so close!

Missing a couple of branches in utils.js, but I've also added a couple of ignores I would like to drop. Help appreciated!

Missing coverage:
image

### CODE REVIEW

File: rules/no-disabled-tests.js [hunk 1]
```diff
@@ -1,19 +1,16 @@
 'use strict';
 
 function getName(node) {
-  function joinNames(a, b) {
```
Comment (SimenB) [inline rules/no-disabled-tests.js]: @borilla mind sharing why you added the check? None of the tests hit this path, so I just inlined it

File: rules/no-disabled-tests.js [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
       return node.name;
     case 'Literal':
       return node.value;
     case 'MemberExpression':
-      return joinNames(getName(node.object), getName(node.property));
+      return `${getName(node.object)}.${getName(node.property)}`;
   }
 
+  /* istanbul ignore next */
   return null;
```
Comment (SimenB) [inline rules/no-disabled-tests.js]: Is this just to be future proof?

File: rules/no-focused-tests.js [hunk 3]
```diff
@@ -27,6 +27,7 @@ module.exports = {
   create: context => ({
     CallExpression(node) {
       const callee = node.callee;
+      /* istanbul ignore if */
       if (!callee) {
```
Comment (SimenB) [inline rules/no-focused-tests.js]: @jkimbo this is an old one, can you provide details on the cases where a `CallExpression` has no `callee`?

File: rules/valid-expect-in-promise.js [hunk 4]
```diff
@@ -17,6 +17,7 @@ const isExpectCallPresentInFunction = body => {
     return body.body.find(line => {
       if (line.type === 'ExpressionStatement')
         return isExpectCall(line.expression);
+      /* istanbul ignore else */
       if (line.type === 'ReturnStatement') return isExpectCall(line.argument);
```
Comment (SimenB) [inline rules/valid-expect-in-promise.js]: @tushardhole can you provide some sample code where this condition is `false`?

**Comment by SimenB** [comment]
@xfumihiro Do you think we could add some tests covering the uncovered parts of `utils.js`?

File: rules/no-disabled-tests.js (hunk 1 above)
Comment (borilla) [reply rules/no-disabled-tests.js]: Just defensive coding, but I can come up with a theoretical example using an anonymous object:
```
({ f: function () {} }).f()
```

File: rules/no-disabled-tests.js (hunk 2 above)
Comment (borilla) [reply rules/no-disabled-tests.js]: This is for any unmatched node-types (ie those we can't find names for), eg
```
(a || b).f(); // callee.object here is LogicalExpression
```

Comment (SimenB) [reply rules/no-disabled-tests.js]: Great, thanks! Added those 2 examples, and coverage hit 100% 🙂

Comment (borilla) [reply rules/no-disabled-tests.js]: :+1:

**Comment by SimenB** [comment]
I'm gonna merge as is, and open up a separate issue for 100%

**Pull request merged**